components/StoreProvider/StoreProvider.tsx

```diff
@@ -0,0 +1,18 @@
+'use client'
+import { AppStore, makeStore } from '@/store/store'
+import { useRef } from 'react'
+import { Provider } from 'react-redux'
+
+export default function StoreProvider({
+    children,
+}: {
+    children: React.ReactNode
+}) {
+    const storeRef = useRef<AppStore>()
```

Comment:
Interesting approach, but why did we do this?

Reply:
This is to separate server component and client component in next js. As store provider only works in use client, we create a HOC and wrap this in server layout.tsx